pkg/mimir/promexts.go

```diff
@@ -3,14 +3,9 @@
 package mimir
 
 import (
-	"github.com/prometheus/common/model"
-
 	"github.com/grafana/mimir/pkg/util/promqlext"
 )
 
 func init() {
 	promqlext.ExtendPromQL()
-	// Mimir doesn't support Prometheus' UTF-8 metric/label name scheme yet.
-	// nolint:staticcheck
-	model.NameValidationScheme = model.LegacyValidation
```

Comment:
I think we need to retain this for Prometheus code that relies on the global variable (eg. Prometheus' query engine).